Program.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace Homework7
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            //Car MyCar = new Car("Ford", "Mustang", 1995, 23156.0f);
+
+            //Console.WriteLine("Drive() Method");
+            //MyCar.Drive(123.35f);
+            //Console.WriteLine(MyCar.WriteOutInfo("mileage"));
+
+            //Console.WriteLine("\n");
+            //Console.WriteLine("DisplayCarInfo() Method");
+            //MyCar.DisplayCarInfo();
+
+            //Console.WriteLine("\n");
+            //Console.WriteLine(MyCar);
+            //Console.WriteLine("Press any key...");
+            //Console.ReadKey();
+
+
+            //Book RandomBook = new Book("The Great Gatsby", "F. Scott Fitzgerald", 180);
+            //RandomBook.Read(50);
+            //RandomBook.DisplayProgress();
+            //Console.WriteLine("Press any key...");
+            //Console.ReadKey();
+
+
+        }
+    }
+
+    public class Car
+    {
+        private string Name { get; } //Pořád si nejsem jistý, kdy je dobrá praxe používat private-public,
+        private string Model { get; } //ale říkal jsem si, že kdybych měl někde v databázi uvedené auté, 
+        private int Year { get; } //tak bych ho nechtěl jen tak změnit. Zvlášť mileage kvůli scamům.
+        private float Mileage { get; set; }
+
+        public Car(string name, string model, int year, float mileage)
+        {
+            Name = name;
+            Model = model;
+            Year = year;
+            Mileage = mileage;
+        }
+
+        public void Drive(float kiloMeters)
+        {
+            if (kiloMeters < 0)
+            {
+                throw new ArgumentException("Kilometers driven cannot be negative.");
+            }
+            this.Mileage = this.Mileage + kiloMeters;
+        }
+        public string WriteOutInfo(string input)
+        {
+            if (input == "name" || input == "Name" || input == "NAME")
+            {
+                return this.Name;
+            }
+            else if (input == "model" || input == "Model" || input == "MODEL")
+            {
+                return this.Model;
+            }
+            else if (input == "year" || input == "Year" || input == "YEAR")
+            {
+                return this.Year.ToString();
+            }
+            else if (input == "mileage" || input == "Mileage" || input == "MILEAGE")
+            {
+                return this.Mileage.ToString();
+            }
+            else
+            {
+                return "EXCEPTION: InputUnknown";
+            }
+
+        }
+        public override string ToString() //Chtěl jsem i jednu metodu, která by přímo vracela string a ne jen vypisovala
```

Comment:
Jo, super napad. Bezne se prave (treba pri zalogovani si stavu systemu) pouziva ToString(). Ten systemovej vetsinou nemusi fungovat spravne (nebo treba by mohl leakovat nejake tajne hodnoty), takze se casto overriduje (stejne jako jsi udelal ty).

Proc to ma kazdy objekt zkusim vysvetlit na pristi lekci. Mam to v poznamkach, ze jsem to pri lekci na classy zapomnel zminit a vysvetlit a tak snad na to v utery nezapomenu 😄

Reply:
A je to super, ze jsi si to vyzkousel. Pak v Mainu je krasne videt, ze kdyz das ten objekt primo do vypisu
`Console.WriteLine(MyCar);` , tak to uz zvladne vypsat spravne.